internal/auth0/user/resource_permission_test.go

```diff
@@ -0,0 +1,128 @@
+package user_test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+
+	"github.com/auth0/terraform-provider-auth0/internal/acctest"
+)
+
+const givenAResourceServerWithPermissions = `
+resource "auth0_resource_server" "resource_server" {
+	name = "Acceptance Test - {{.testName}}"
+	identifier = "https://uat.api.terraform-provider-auth0.com/{{.testName}}"
+	scopes {
+		value = "read:foo"
+		description = "Can read Foo"
+	}
+	scopes {
+		value = "create:foo"
+		description = "Can create Foo"
+	}
+}
+`
+
+const givenAUserPermission = `
+resource "auth0_user_permission" "user_permission_read" {
+	depends_on = [auth0_resource_server.resource_server, auth0_user.user ]
+
+	user_id = auth0_user.user.id
+	resource_server_identifier = auth0_resource_server.resource_server.identifier
+	permission = "read:foo"
+}
+`
+
+const givenAnotherUserPermission = `
+resource "auth0_user_permission" "user_permission_create" {
+	depends_on = [auth0_resource_server.resource_server, auth0_user.user ]
+
+	user_id = auth0_user.user.id
+	resource_server_identifier = auth0_resource_server.resource_server.identifier
+	permission = "create:foo"
+}
+`
+
+const givenAUserWithNoPermissions = testAccUserEmpty
+const givenAUserWithOnePermission = givenAUserWithNoPermissions + givenAUserPermission
+const givenAUserWithTwoPermissions = givenAUserWithOnePermission + givenAnotherUserPermission
+
+const testAccUserPermissionNoneAssigned = givenAResourceServerWithPermissions + givenAUserWithNoPermissions + `data "auth0_user" "user" {
```

Comment:
We don't need to use a data source to retrieve the user details, we already have access to the user resource.